tests/Controllers/Multiplayer/RoomsControllerTest.php

```diff
@@ -429,6 +429,100 @@ public function testJoinWithPassword()
         $this->assertSame($initialUserChannelCount + 1, UserChannel::count());
     }
 
+    public function testDestroy()
+    {
+        $token = Token::factory()->create(['scopes' => ['*']]);
```

Comment:
there's `$this->actAsScopedUser($room->user)`

Reply:
I'm not sure how to use that. I assumed it'd be something like

```diff
diff --git a/tests/Controllers/Multiplayer/RoomsControllerTest.php b/tests/Controllers/Multiplayer/RoomsControllerTest.php
index 7cd7096c82..ae457dcb9d 100644
--- a/tests/Controllers/Multiplayer/RoomsControllerTest.php
+++ b/tests/Controllers/Multiplayer/RoomsControllerTest.php
@@ -431,11 +431,9 @@ class RoomsControllerTest extends TestCase
 
     public function testDestroy()
     {
-        $token = Token::factory()->create(['scopes' => ['*']]);
         $start = now();
         $end = $start->clone()->addMinutes(60);
         $room = Room::factory()->create([
-            'user_id' => $token->user,
             'starts_at' => $start,
             'ends_at' => $end,
             'type' => Room::PLAYLIST_TYPE,
@@ -443,8 +441,8 @@ class RoomsControllerTest extends TestCase
         $end = $room->ends_at; // creation truncates fractional second part, so refetch here
         $url = route('api.rooms.destroy', ['room' => $room]);
 
+        $this->actAsScopedUser($room->user, ['lazer']);
         $this
-            ->actingWithToken($token)
             ->delete($url)
             ->assertSuccessful();
 

```

but trying that, it barfs at me with `App\Exceptions\InvalidScopeException: * is not allowed with Client Credentials`.

Reply:
oh wow the model doesn't have user relation defined (and thus it passes `null` for the user) o_O

adding it is probably a good idea...

Reply:
I think it might be `->host` actually. Seems to work locally at least.

https://github.com/ppy/osu-web/blob/1fbc73baa8e8be6759b1cc4f4ab509d8ae53a165/app/Models/Multiplayer/Room.php#L557